tests/ProxyTest.php

```diff
@@ -135,4 +137,18 @@ public function testCreateObjectProxyFromClass(): void
 
         $this->assertSame(2, $object->edgesCount());
     }
+
+    /**
+     * @link https://github.com/yiisoft/proxy/issues/26
+     * @link https://github.com/yiisoft/proxy/issues/45
+     */
+    public function testClassWithBuiltInInterface(): void
+    {
+        $path = sys_get_temp_dir();
```

Comment:
https://github.com/yiisoft/proxy/issues/42

Reply:
I think it's better to handle it in a separate PR. I already checked it and it's not that easy. At least `php://memory` does not allow exclusive locks and subdirectories.